src/main/fc/config.c

```diff
@@ -527,30 +528,43 @@ bool readEEPROM(void)
     return success;
 }
 
-void writeEEPROM(void)
+static void _writeEEPROM(bool setConfigured)
 {
     validateAndFixConfig();
 
     suspendRxPwmPpmSignal();
 
+    if (setConfigured) {
+        systemConfigMutable()->configured = 1;
+    }
+
     writeConfigToEEPROM();
 
     resumeRxPwmPpmSignal();
 }
 
+void writeEEPROM(void)
+{
+    _writeEEPROM(true);
+}
+
 void writeEEPROMWithFeatures(uint32_t features)
 {
     featureDisableAll();
     featureEnable(features);
 
-    writeEEPROM();
+    _writeEEPROM(true);
 }
 
 void resetEEPROM(void)
 {
     resetConfigs();
 
-    writeEEPROM();
+#ifdef GENERIC_TARGET
```

Comment:
I'd turn this around:
- introduce `USE_CONFIGURATION_STATE` or similar conditional for indirection;
- only touch `configured` if `USE_CONFIGURATION_STATE` is defined, leave false otherwise;
- introduce `bool getSystemConfigured(void)` getter, hardcode it to return `true` if `USE_CONFIGURATION_STATE` isn't defined, `configured` otherwise.

Cleaner in my opinion, and in line with how other parameters are dealt with when the feature using them isn't defined.

Reply:
Accepted.